File: tarefa008/Cidade.ts
import { Estado } from "./Estado";

export class Cidade {
  constructor(nome:string) {
    this._nome = nome;
  }

  //atributos
  private _nome:string;
  private _estado:Estado;

  //setters
  set estado(novoEstado:Estado) {
    this.estado = novoEstado;
  }
}

File: tarefa008/java/Cidade.java
package tarefa008.java;

/**
 * Classe que representa a entidade Cidade |
 * Autor: João Máiro |
 * Data: 15/07/22
 */
public class Cidade {
  Cidade(String nome) {
    this.nome = nome;
  }

  //atributos
  private String nome;
  private Estado estado;

  /**
   * Método que modifica o valor do atributo estado
   * @param estado
   */
  public void setEstado(Estado estado) {
    this.estado = estado;
  }

  /**
   * Método que retorna o valor do atributo nome
   * @return
   */
  public String getNome() {
    return this.nome;
  }

  /**
   * Método que retorna o valor do atributo estado
   * @return
   */
  public Estado getEstado() {
    return this.estado;
  }
}
